OpenRA.Mods.Common/Traits/Cargo.cs

```diff
@@ -144,7 +144,12 @@ void INotifyCreated.Created(Actor self)
 			conditionManager = self.Trait<ConditionManager>();
 		}
 
-		static int GetWeight(Actor a) { return a.Info.TraitInfo<PassengerInfo>().Weight; }
+		public int GetWeight(Actor a) { return GetPassengerTrait(a).Info.Weight; }
+
+		public Passenger GetPassengerTrait(Actor a)
+		{
+			return a.TraitsImplementing<Passenger>().Single(pa => Info.Types.Contains(pa.Info.CargoType));
```

Comment:
This will throw at runtime if the unit has multiple matching passenger traits.  I would prefer that you enumerated and acted on all the matching traits, but failing that you should write a lint test to statically detect this.